app/main.py

```diff
@@ -1,3 +1,65 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
```

Comment:
The `__add__` method should raise an exception if `other` is not a `Distance` instance and not a number. Currently, it assumes `other` is a number, which might not always be the case.